client/src/components/dashboard/appointments/Appointments.js

```diff
@@ -6,13 +6,13 @@ export default class Appointments extends PureComponent {
   static propTypes = {
     appointments: PropTypes.array.isRequired,
     fetchAppointments: PropTypes.func.isRequired,
-    user: PropTypes.object.isRequired,
-    match: PropTypes.object.isRequired
+    user: PropTypes.object.isRequired
   };
 
   componentDidMount() {
     const { fetchAppointments } = this.props;
-    const userId = this.props.match.params.id;
+    // Fetch appointments for the session user
+    const userId = user._id;
     fetchAppointments(userId);
```

Comment:
Get the userId from the session instead of the URL bar. A user should not be able to get other users appointments, so there is no need to have the userId in the url